proposals/partial-type-inference.md

```diff
@@ -0,0 +1,743 @@
+# Partial type inference
+
+* [x] Proposed
```

Comment:
Those bullets are best kept in the championed proposal ([https://github.com/dotnet/csharplang/issues/1349](https://github.com/dotnet/csharplang/issues/1349)) so they can be updated fluidly

Reply:
I made some edits directly to save time. I hope that's okay with you.
I removed these bullets and simplified the links in the copied sections of the spec (makes it easier to review the diff).